src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ecc.c

```diff
@@ -3,55 +3,47 @@
 
 #include "pal_ecc.h"
 
-#if !defined(TARGET_MACCATALYST) && !defined(TARGET_IOS) && !defined(TARGET_TVOS)
-int32_t AppleCryptoNative_EccGenerateKey(
-    int32_t keySizeBits, SecKeychainRef tempKeychain, SecKeyRef* pPublicKey, SecKeyRef* pPrivateKey, int32_t* pOSStatus)
+int32_t AppleCryptoNative_EccGenerateKey(int32_t keySizeBits,
+                                         SecKeyRef* pPublicKey,
+                                         SecKeyRef* pPrivateKey,
+                                         CFErrorRef* pErrorOut)
 {
     if (pPublicKey != NULL)
         *pPublicKey = NULL;
     if (pPrivateKey != NULL)
         *pPrivateKey = NULL;
 
-    if (pPublicKey == NULL || pPrivateKey == NULL || pOSStatus == NULL)
+    if (pPublicKey == NULL || pPrivateKey == NULL || pErrorOut == NULL)
         return kErrorBadInput;
 
+    int32_t ret = kErrorSeeError;
     CFMutableDictionaryRef attributes = CFDictionaryCreateMutable(NULL, 3, &kCFTypeDictionaryKeyCallBacks, NULL);
-
     CFNumberRef cfKeySizeValue = CFNumberCreate(NULL, kCFNumberIntType, &keySizeBits);
-    OSStatus status;
 
     if (attributes != NULL && cfKeySizeValue != NULL)
     {
         CFDictionaryAddValue(attributes, kSecAttrKeyType, kSecAttrKeyTypeEC);
         CFDictionaryAddValue(attributes, kSecAttrKeySizeInBits, cfKeySizeValue);
-        CFDictionaryAddValue(attributes, kSecUseKeychain, tempKeychain);
```

Comment:
I vaguely feel like part of this temporary keychain stuff is because the macOS APIs would add things to the default keychain by default.  Is that no longer the case with the new API (perhaps that's the permanent: true vs permanent: false, thing?)?

Reply:
It's the principal difference between `SecCreateRandomKey` and `SecKeyGenerateKeyPair`. `SecCreateRandomKey` doesn't default to permanent keys and user's keychain.